lib/spack/docs/getting_started.rst

```diff
@@ -1514,3 +1514,203 @@ To ensure that Spack does not autodetect the Cray programming
 environment, unset the environment variable ``MODULEPATH``. This
 will cause Spack to treat a linux container on a Cray system as a base
 linux distro.
+
+.. _windows_support:
+
+----------------
+Spack On Windows
+----------------
+
+Windows support for Spack is currently under development.  While this work is still in an early stage,
+it is currently possible to set up Spack and perform a few operations on Windows.  This section will guide
+you through the steps needed to install Spack and start running it on a fresh Windows machine. 
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Step 1: Install prerequisites
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To use Spack on Windows, you will need the following packages:
+
+* Microsoft Visual Studio
+* Intel Fortran (needed for some packages)
+* Python 
+* Git
+
+"""""""""""""""""""""""
+Microsoft Visual Studio
+"""""""""""""""""""""""
+
+Microsoft Visual Studio provides the Windows C/C++ compiler that is currently supported by Spack.
+
+We require several specific components to be included in the Visual Studio installation.
+One is the C/C++ toolset, which can be selected as "Desktop development with C++" or "C++ build tools,"
+depending on installation type (Professional, Build Tools, etc.)  The other required component is
+"C++ CMake tools for Windows," which can be selected from among the optional packages.
+This provides CMake and Ninja for use during Spack configuration.
+
+If you already have Visual Studio installed, you can make sure these components are installed by
+rerunning the installer.  Next to your installation, select "Modify" and look at the
+"Installation details" pane on the right.
+
+"""""""""""""
+Intel Fortran
+"""""""""""""
+
+For Fortran-based packages on Windows, we strongly recommend Intel's oneAPI Fortran compilers.
+The suite is free to download from Intel's website, located at 
+https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/fortran-compiler.html#gs.70t5tw.
+The executable of choice for Spack will be Intel's Beta Compiler, ifx, which supports the classic
+compiler's (ifort's) frontend and runtime libraries by using LLVM.
```

Comment:
(this should be work for a future PR but want to check in if this makes sense here)

I think we should be able to `spack install intel-oneapi-compilers` after adding MSVC (at which point we should be able to ``spack compiler add `spack location -i intel-oneapi-compilers`/bin``. 

To do that, we'll need to add a Windows-compatible version of oneAPI to `intel-oneapi-compilers` which follows the same model as Linux:

```
    if platform.system() == 'Linux':
        version('2022.0.1',
                url='https://registrationcenter-download.intel.com/akdlm/irc_nas/18435/l_dpcpp-cpp-compiler_p_2022.0.1.71_offline.sh',
                sha256='c7cddc64c3040eece2dcaf48926ba197bb27e5a46588b1d7b3beddcdc379926a',
                expand=False)
        resource(name='fortran-installer',
                 url='https://registrationcenter-download.intel.com/akdlm/irc_nas/18436/l_fortran-compiler_p_2022.0.1.70_offline.sh',
                 sha256='2cb28a04f93554bfeffd6cad8bd0e7082735f33d73430655dea86df8933f50d1',
                 expand=False,
                 placement='fortran-installer',
                 when='@2022.0.1')
```


Also, since there isn't actually any compilation that is done with this, we might also get away with defining a "noop" compiler that we create in the `compilers.yaml` config on Windows: we could use that to do an initial installation of `intel-oneapi-compilers`. I think that would be a huge win since we then wouldn't depend on MSVC for low-hassle definition of a working compiler (this assumes that `icx` is usable on Windows).

Even if that were fixed, these instructions appear to currently make use of MSVC for getting CMake/Ninja easily. It looks like we could download Ninja from https://github.com/ninja-build/ninja/releases (and likewise you can download CMake), at which point `spack external find` could check those download directories.

Reply:
There was much going back and forth about where and how to get CMake with @tgamblin and others. It was decided that downloading it was not good in the long run. So, we can now build CMake. You can also build ninja at this point.  There are some bootstrapping issues with CMake. In a future PR we will have CMake binaries hosted on a spack server and get them from there. For now it can bootstrap CMake and we can build ninja from source.